checker-qual/src/main/java/org/checkerframework/checker/mustcall/qual/MustCallAlias.java

```diff
@@ -0,0 +1,61 @@
+package org.checkerframework.checker.mustcall.qual;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This polymorphic annotation represents an either-or must-call obligation. This annotation should
+ * always be used in pairs: one on a method receiver parameter or constructor parameter, and the
+ * other on the method return type. Fulfilling the must-call obligation of one is equivalent to
+ * fulfilling the must-call obligation of the other.
+ *
+ * <p>This annotation is useful for wrapper objects. For example, consider the declaration of {@code
+ * java.net.Socket#getOutputStream}:
+ *
+ * <pre>
+ * &#64;MustCall("close")
+ * class Socket {
+ *   &#64;MustCallAlias OutputStream getOutputStream(&#64;MustCallAlias Socket this) { ... }
+ * }
+ * </pre>
+ *
+ * When a client calls {@code getOutputStream} on the Socket creates a new {@code OutputStream}
+ * object that wraps the Socket. Calling its {@code close()} method will close the underlying
+ * socket, but the Socket may also be closed directly. Calling close on either object is permitted
+ * -- thus, both the receiver and return type above are annotated as {@code @MustCallAlias}.
+ *
+ * <h3>Verifying {@code @MustCallAlias} annotations</h3>
+ *
+ * Suppose that {@code @MustCallAlias} is written on the type of parameter {@code p}.
+ *
+ * <p>For a constructor:
+ *
+ * <ul>
+ *   <li>The constructor must always write p into exactly one field {@code f} of the new object.
+ *   <li>Field {@code f} must be annotated {@code @}{@link Owning} (which necessitates that a
+ *       must-call method of the object being constructed resolves's {@code f}'s must-call
```

Comment:
Yes, which is a general property of `@Owning` fields that the consistency checker enforces. I think it's redundant here, though, and the parenthesis seems to be causing rather than relieving confusion, so I'm just going to remove it. A user can look up the rules for `@Owning` fields separately in the consistency checker's documentation.